### PR TITLE
switch to testthat 3rd edition and adapt tests

### PR DESCRIPTION
**Pull request by IndrajeetPatil**
Expect a few tests to fail due to changes in how `testthat` now checks for values. 

I have changed as many tests as I can, but some I have not because I am not sure if it is a good idea for me to be changing them without knowing the full context in which they were written.

### CODE REVIEW

File: tests/testthat/test-brms.R [hunk 1]
```diff
@@ -74,7 +74,7 @@ if (require("brms") && require("testthat") && require("insight") && require("htt
 
     out <- describe_posterior(model, effects = "all", components = "all", centrality = "mean", test = NULL)
     s <- summary(model)
-    expect_equal(s$fixed[, 1, drop = TRUE], out$Mean, check.attributes = FALSE, tolerance = 1e-3)
-    expect_equal(s$fixed[, 5, drop = TRUE], out$Rhat, check.attributes = FALSE, tolerance = 1e-1)
```
Comment (strengejacke) [inline tests/testthat/test-brms.R:78]: `as.vector()` works here, in other situations, where we have no vector, the new argument is `ignore_attr = TRUE`.

**Comment by strengejacke** [comment]
> but some I have not because I am not sure if it is a good idea for me to be changing them without knowing the full context in which they were written.

Can you give an example?

**Comment by IndrajeetPatil** [comment]
Here are some of the failures:

```r
Failure (test-p_map.R:4:5): p_map
as.numeric(p_map(distribution_normal(1000, 1, 1))) (`actual`) not equal to 0.62 (`expected`).

  `actual`: 0.63
`expected`: 0.62

Failure (test-p_map.R:5:5): p_map
as.numeric(p_map(distribution_normal(1000, 2, 1))) (`actual`) not equal to 0.15 (`expected`).

  `actual`: 0.16
`expected`: 0.15

Failure (test-weighted_posteriors.R:38:5): weighted_posteriors for BayesFactor (intercept)
wHDI$CI_low (`actual`) not equal to c(-0.519, -0.64) (`expected`).

  `actual`: -0.519 -0.641
`expected`: -0.519 -0.640

Failure (test-weighted_posteriors.R:39:5): weighted_posteriors for BayesFactor (intercept)
wHDI$CI_high (`actual`) not equal to c(0.15, 0.059) (`expected`).

actual != expected but don't know how to show the difference

Failure (test-density_at.R:2:3): density_at
density_at(distribution_normal(1000), 0) (`actual`) not equal to 0.389 (`expected`).

  `actual`: 0.38
`expected`: 0.39

Failure (test-density_at.R:3:3): density_at
density_at(distribution_normal(1000), c(0, 1))[1] (`actual`) not equal to 0.389 (`expected`).

  `actual`: 0.38
`expected`: 0.39
```

**Comment by strengejacke** [comment]
p_map and density also fail on GitHub CI:
https://github.com/easystats/bayestestR/pull/369/checks?check_run_id=1790533240

**Comment by mattansb** [comment]
The values for the weighted posterior tests can be fixed accordingly. (Let me know if you prefer I do it)

**Comment by IndrajeetPatil** [comment]
I won't be making any more changes in this PR, so you can either push changes to fix breaking tests in the PR itself or fix them once the PR is merged.

**Pull request merged**